Libraries/CO.CDP.MQ/Sqs/SqsDispatcher.cs

```diff
@@ -0,0 +1,102 @@
+using Amazon.SQS;
+using Amazon.SQS.Model;
+
+namespace CO.CDP.MQ.Sqs;
+
+public delegate object Deserializer(string type, string body);
+
+public delegate bool TypeMatcher(Type type, string typeName);
+
+public record SqsDispatcherConfiguration
+{
+    public required string QueueName { get; init; }
+    public required int MaxNumberOfMessages { get; init; } = 1;
+    public required int WaitTimeSeconds { get; init; } = 30;
+}
+
+public class SqsDispatcher(
+    AmazonSQSClient sqsClient,
+    SqsDispatcherConfiguration configuration,
+    Deserializer deserializer,
+    TypeMatcher typeMatcher)
+    : IDispatcher
+{
+    public SqsDispatcher(
+        AmazonSQSClient sqsClient, SqsDispatcherConfiguration configuration, Deserializer deserializer)
+        : this(sqsClient, configuration, deserializer, (type, typeName) => type.Name == typeName)
+    {
+    }
+
+    private const string TypeAttribute = "Type";
+    private readonly Dictionary<Type, List<Func<object, Task>>> _subscribers = [];
```

Comment:
Should this be ConcurrentDictionary ?

Reply:
Good question! I considered it, but couldn’t think of a scenario to justify it. Can you think of one?

Reply:
Let’s keep this mind and change to concurrent connection once we prove it’s needed.